### PR TITLE
Let simulation start server with custom ClientManager

### PR DESCRIPTION
**Pull request by negedng**
#### Reference Issues/PRs
Can't start simulation with custom ClientManager.

#### What does this implement/fix? Explain your changes.
Adding extra parameters to default server initialization and simulation start to let users add a custom ClientManager instead of always using the SimpleClientManager.

### CODE REVIEW

**Comment by tanertopal** [comment]
@negedng thank you for the PR! Could you additionally prepare a changelog entry?

@danieljanes this looks good to me. I don't see any downsides of passing a custom client manager. Wdyt?

**Comment by negedng** [comment]
@tanertopal I'm not sure what do you mean by adding a changelog entry. I updated the `doc/source/changelog.rst`, I hope it is the right way.

Also, one of the tests showed a formatting issue that I changed. I hope it is better now!

File: src/py/flwr/simulation/app.py [hunk 1]
```diff
@@ -96,6 +100,10 @@ def start_simulation(  # pylint: disable=too-many-arguments
         An implementation of the abstract base class `flwr.server.Strategy`. If
         no strategy is provided, then `start_server` will use
         `flwr.server.strategy.FedAvg`.
+    client_manager: Optional[flwr.server.ClientManager] (default: None)
+        An implementation of the abstract base class `flwr.server.ClientManager`.
+        If no strategy is provided, then `start_server` will use
```
Comment (danieljanes) [inline src/py/flwr/simulation/app.py:105]:
```suggestion
        If no implementation is provided, then `start_simulation` will use
```

File: src/py/flwr/simulation/app.py [hunk 2]
```diff
@@ -24,8 +24,10 @@
 from flwr.client.client import Client
 from flwr.common.logger import log
 from flwr.server.app import _fl, _init_defaults
+from flwr.server.client_manager import ClientManager
 from flwr.server.history import History
 from flwr.server.strategy import Strategy
+
```
Comment (danieljanes) [inline src/py/flwr/simulation/app.py:30]: Should this empty line be removed?

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Comment by danieljanes** [comment]
Thanks for the PR @negedng, looks good! I've just added two small comments.

@tanertopal I agree, let's merge this :)

**Comment by tanertopal** [comment]
@negedng do you want to apply Daniels suggestions or shall I do? Afterwards we can merge it.

**Comment by negedng** [comment]
Also, a test shows that there is a sorting issue in the imports for simulation/app.py, but I can't figure out the proper way to sort the imports. Can you have a look? thanks!

File: src/py/flwr/simulation/app.py [hunk 3]
```diff
@@ -149,7 +156,7 @@ def start_simulation(  # pylint: disable=too-many-arguments
 
     # Initialize server and server config
     config = {"num_rounds": num_rounds}
-    initialized_server, initialized_config = _init_defaults(None, config, strategy)
+    initialized_server, initialized_config = _init_defaults(None, config, strategy, client_manager)
```
Comment (tanertopal) [inline src/py/flwr/simulation/app.py:159]:
```suggestion
    initialized_server, initialized_config = _init_defaults(
        None, config, strategy, client_manager
    )
```

**Comment by negedng** [comment]
Hi, any update on this? Do I need to change something?

**Comment by tanertopal** [comment]
@negedng There was an issue with PyPi at the time I wanted to merge it and CI got blocked iirc. I'll have a look today again and merge it if the CI goes through.

**Review by tanertopal — APPROVED** [review]

**Comment by tanertopal** [comment]
@negedng there is still an issue in:

```src/py/flwr/server/app.py:97: error: Missing positional argument "client_manager" in call to "_init_defaults"```

could you have a look?

**Review by tanertopal — APPROVED** [review]

File: src/py/flwr/server/app.py [hunk 4]
```diff
@@ -55,6 +56,10 @@ def start_server(  # pylint: disable=too-many-arguments
             implementation of the abstract base class `flwr.server.Strategy`.
             If no strategy is provided, then `start_server` will use
             `flwr.server.strategy.FedAvg`.
+        client_manager: Optional[flwr.server.ClientManager] (default: None)
+            An implementation of the abstract base class `flwr.server.ClientManager`.
+            If no implementation is provided, then `start_simulation` will use
```
Comment (tanertopal) [inline src/py/flwr/server/app.py:61]:
```suggestion
            If no implementation is provided, then `start_server` will use
```

**Review by tanertopal — CHANGES_REQUESTED** [review]

**Review by tanertopal — DISMISSED** [review]

File: src/py/flwr/server/app.py [hunk 5]
```diff
@@ -94,7 +99,7 @@ def start_server(  # pylint: disable=too-many-arguments
     >>>     )
     >>> )
     """
-    initialized_server, initialized_config = _init_defaults(server, config, strategy)
+    initialized_server, initialized_config = _init_defaults(server, config, strategy, client_manager)
```
Comment (tanertopal) [inline src/py/flwr/server/app.py:102]:
```suggestion
    initialized_server, initialized_config = _init_defaults(
        server, config, strategy, client_manager
    )
```

**Review by tanertopal — APPROVED** [review]

**Pull request merged**